tests/TwigRuntimeExtensionTest.php

```diff
@@ -206,19 +262,70 @@ public function testUrlFor(
         $this->assertEquals($expectedUrl, $extension->urlFor($routeName, $routeData, $queryParams));
     }
 
+    public function relativeUrlForProvider()
+    {
+        return [
+             ['',           '/user/1/',  '',    '/user/{id}/',     ['id' => 1], [],           ''],
+             ['/base-path', '/user/1/',  'a=b', '/user/{id}/edit', ['id' => 1], [],           'edit'],
+             ['',           '/user/1/',  'a=b', '/user/',          [],          ['s' => 'n'], '../?s=n'],
+             ['',           '/user/',    's=a', '/user/',          [],          ['s' => 'b'], '?s=b'],
+             ['/base-path', '/user/add', '',    '/user/',          [],          ['p' => 2],   './?p=2'],
+        ];
```

Comment:
If we move the code of `\Slim\Views\TwigRuntimeExtension::relativeUrl` to `\Slim\Views\TwigRuntimeExtension::relativeUrlFor`, then we might put all the cases from `\Slim\Tests\TwigRuntimeExtensionTest::relativeUrlProvider` to this provider:

So for example 
```php
['/a',                  '/a',                   ''],
```
becomes
```php
['', '/a', '', '/a', [], [], ''],
```

Here are the first few (just in case):
```php
// ['/a',                  '/a?b=c',               'a'],
   ['', '/a?b=c', '', '/a', [], [], 'a'],
// ['/a?b=c',              '/a',                   '?b=c'],
   ['', '/a', '', '/a', [], ['b' => 'c'], '?b=c'],
// ['/a?b=c',              '/a?b=c',               ''],
   ['', '/a?b=c', '', '/a', [], ['b' => 'c'], ''],
// ['/a?b=c',              '/a?b=d',               '?b=c'],
   ['', '/a?b=d', '', '/a', [], ['b' => 'c'], '?b=c'],
```